### PR TITLE
1.2.0 release

### PR DESCRIPTION
**Pull request by StefanBratanov**

### CODE REVIEW

**Review by tbenr — APPROVED** [review]
LGTM

File: libp2p/src/main/kotlin/io/libp2p/discovery/MDnsDiscovery.kt [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
 import io.libp2p.discovery.mdns.AnswerListener
 import io.libp2p.discovery.mdns.JmDNS
 import io.libp2p.discovery.mdns.ServiceInfo
 import io.libp2p.discovery.mdns.impl.DNSRecord
 import io.libp2p.discovery.mdns.impl.constants.DNSRecordType
-import java.net.Inet4Address
-import java.net.Inet6Address
-import java.net.InetAddress
+import java.net.*
+import java.util.*
```
Comment (tbenr) [inline libp2p/src/main/kotlin/io/libp2p/discovery/MDnsDiscovery.kt:13]: wonder why we have this wildcard imports

File: libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipParams.kt [hunk 2]
```diff
@@ -231,7 +231,24 @@ data class GossipParams(
...
+    /**
+     * [maxIDontWantMessageIds] is the maximum number of IDONTWANT message ids allowed per heartbeat per peer
+     */
+    val maxIDontWantMessageIds: Int = maxIHaveLength * maxIHaveMessages,
+
+    /**
+     * [iDontWantMinMessageSizeThreshold] controls the minimum size (in bytes) that an incoming message needs to be so that an IDONTWANT message is sent to mesh peers.
+     * The default is 16 KB.
+     */
+    val iDontWantMinMessageSizeThreshold: Int = 16000,
```
Comment (tbenr) [inline libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipParams.kt:245]: I know it doesn't matter but my mind works with 16384 :)

Comment (Nashatyrev) [reply libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipParams.kt:245]: Haha, was going to leave the same note, but strictly technically speaking 16 Kilobytes = 16000 bytes 😆

**Review by Nashatyrev — APPROVED** [review]
LGTM

File: libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipParams.kt (hunk 2 above)
Comment (tbenr) [reply libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipParams.kt:245]: yeah `KB` is stated not `KiB`, so it is consistent :-)

**Pull request merged**